doc/source/raysgd/v2/user_guide.rst

```diff
@@ -1,167 +1,311 @@
 
-    def train_simple(config: Dict):
+    **Step 1:** Wrap your model in ``MultiWorkerMirroredStrategy``.
 
-        # N is batch size; D_in is input dimension;
-        # H is hidden dimension; D_out is output dimension.
-        N, D_in, H, D_out = 8, 5, 5, 5
+    The `MultiWorkerMirroredStrategy <https://www.tensorflow.org/api_docs/python/tf/distribute/experimental/MultiWorkerMirroredStrategy>`_
+    enables synchronous distributed training. The ``Model`` *must* be built and
+    compiled within the scope of the strategy.
 
-        # Create random Tensors to hold inputs and outputs
-        x = torch.randn(N, D_in)
-        y = torch.randn(N, D_out)
-        loss_fn = nn.MSELoss()
+    .. code-block:: python
 
-        # Use the nn package to define our model and loss function.
-        model = torch.nn.Sequential(
-            torch.nn.Linear(D_in, H),
-            torch.nn.ReLU(),
-            torch.nn.Linear(H, D_out),
-        )
-        optimizer = optim.SGD(model.parameters(), lr=0.1)
+        with tf.distribute.MultiWorkerMirroredStrategy().scope():
+            model = ... # build model
+            model.compile()
 
-        model = DistributedDataParallel(model)
-        results = []
+    **Step 2:** Update your ``Dataset`` batch size to the *global* batch
+    size.
 
-        for epoch in range(config.get("epochs", 10)):
-            optimizer.zero_grad()
-            output = model(x)
-            loss = loss_fn(output, y)
-            loss.backward()
-            results.append(loss.item())
-            optimizer.step()
-        return results
+    The `batch <https://www.tensorflow.org/api_docs/python/tf/data/Dataset#batch>`_
+    will be split evenly across worker processes, so ``batch_size`` should be
+    set appropriately.
+
+    .. code-block:: diff
+
+        -batch_size = worker_batch_size
+        +batch_size = worker_batch_size * num_workers
+
+  .. group-tab:: Horovod
 
-Running this with RaySGD is as simple as the following:
+    If you have a training function that already runs with the `Horovod Ray
+    Executor <https://horovod.readthedocs.io/en/stable/ray_include.html#horovod-ray-executor>`_,
+    you should not need to make any additional changes!
+
+    To onboard onto Horovod, please visit the `Horovod guide
+    <https://horovod.readthedocs.io/en/stable/index.html#get-started>`_.
+
+Create RaySGD Trainer
+~~~~~~~~~~~~~~~~~~~~~
+
+The ``Trainer`` is the primary RaySGD class that is used to manage state and
+execute training. You can create a simple ``Trainer`` for the backend of choice
+with one of the following:
 
 .. code-block:: python
 
-    all_results = trainer.run(train_simple)
+    torch_trainer = Trainer(backend="torch", num_workers=2)
 
+    tensorflow_trainer = Trainer(backend="tensorflow", num_workers=2)
 
+    horovod_trainer = Trainer(backend="horovod", num_workers=2)
 
-Porting code to RaySGD
-----------------------
+For more configurability, please reference the :ref:`sgd-api-trainer` API.
```

Comment:
Nit: Can you also link to the `BackendConfig` section in the API docs somewhere here on how to configure the backend.

Reply:
Good call out! I had this before in the code block but accidentally deleted it while cleaning this section up without adding a link...